client/src/main/java/io/split/engine/segments/SegmentSynchronizationTaskImp.java

```diff
@@ -137,6 +141,7 @@ public boolean isRunning() {
 
     public void fetchAll(boolean addCacheHeader) {
         _splitCacheConsumer.getSegments().forEach(this::initialize);
+        _ruleBasedSegmentCacheConsumer.getSegments().forEach(this::initialize);
```

Comment:
I suggest creating a `HashSet` with the names of the segments to ensure they are unique, since the same segment name might appear in both the flags and the RBs.

```java
private HashSet<String> GetSegmentNames() {
        HashSet<String> names = new HashSet<>(_splitCacheConsumer.getSegments());
        names.addAll(_rbsCacheConsumer.getSegments());

        return names;
    }
```